identityserver.plugins.authenticationactions.redirect-action/src/main/java/io/curity/identityserver/plugin/RedirectAction/RedirectActionAuthenticationAction.java

```diff
@@ -0,0 +1,149 @@
+    private final WebServiceClient _client;
+    private final Json _jsonService;
+    private final SessionManager _sessionManager;
+    private final String _redirectURI;
+
+    public RedirectActionAuthenticationAction(RedirectActionAuthenticationActionConfig configuration) throws URISyntaxException
+    {
+        _client = getWebServiceClient(configuration, configuration.getRedirectURL() + configuration.getResultPath());
+        _jsonService = configuration.getJsonService();
+        _sessionManager = configuration.getSessionManager();
+        _redirectURI = configuration.getRedirectURL() + configuration.getRedirectPath() + "?status=";
+    }
+
+    @Override
+    public AuthenticationActionResult apply(AuthenticationAttributes authenticationAttributes,
+                                            AuthenticatedSessions authenticatedSessions,
+                                            String authenticationTransactionId,
+                                            AuthenticatorDescriptor authenticatorDescriptor)
+    {
+        String key = authenticationAttributes.getSubject() + authenticatorDescriptor.getId();
+
+        Attribute parameterFromSession = _sessionManager.get(key);
+
+        if (parameterFromSession == null)
+        {
+            return redirectResponse(key);
+        }
+        else
+        {
+            HttpResponse response = getExternalStatus(parameterFromSession);
+
+            if (response.statusCode() != 200)
+            {
+                return AuthenticationActionResult.failedResult("Couldn't verify external status");
+            }
+
+            Map<String, Object> responseJson = response.body(HttpResponse.asJsonObject(_jsonService));
+
+            String externalUserId = (String) responseJson.get("externalUserId");
+            String externalStatus = (String) responseJson.get("externalStatus");
+
+            _logger.debug("Adding data to authentication: {}, {}", externalUserId, externalStatus);
+            authenticationAttributes.append(Attribute.of("externalUserId", externalUserId));
+            authenticationAttributes.append(Attribute.of("externalStatus", externalStatus));
+
+            return AuthenticationActionResult.successfulResult(authenticationAttributes);
+        }
+
+
+    }
+
+    private HttpResponse getExternalStatus(Attribute parameterFromSession)
+    {
+        return _client.withQuery("status=" + parameterFromSession.getValueOfType(String.class)).request().get().response();
+    }
+
+    private AuthenticationActionResult redirectResponse(String key)
+    {
+        UUID uuid = UUID.randomUUID();
+        _sessionManager.put(Attribute.of(key, uuid.toString()));
+
+        try
+        {
+            URI redirectUri = new URI(_redirectURI + uuid);
+            return AuthenticationActionResult.pendingResult(redirect(redirectUri));
+        }
+        catch (URISyntaxException e)
+        {
+            _logger.warn("Could not construct URI for URL: {}", _redirectURI + uuid);
```

Comment:
can the admin fix this without calling support? If not, this seems like an error to me.

Reply:
This will only happen if the admin has misconfigured the plugin. The admin must have entered as the redirect URL and / or path values which do not make a valid URI. The admin should be able to fix this.